### ISSUE TITLE
Consolidate how Crafty options are set

### ISSUE THREAD

**Issue by starwed** [craftyjs/Crafty#1026]
This is a bit piecemeal right now -- for instance, why is  `Crafty.pixelart` a method?  We should make all the various options properties of some `Crafty.options` object, allow overriding defaults when calling `Crafty.init`, and fire events when the options change.

Not sure whether we want to support nested options, or just require that they all be top-level to simplify things.  (I think to easily work with events, they really need to be top-level only.)

**Comment by mucaho** [comment]
> Not sure whether we want to support nested options, or just require that they all be top-level to simplify things. (I think to easily work with events, they really need to be top-level only.)

I don't understand difference between top-level or nested options. Is there an implementation caveat when trying to merge nested options?
As for events, we could do something similar to what [`Model`](http://craftyjs.com/api/Model.html) does perhaps?

---

On the other hand, what options, beside `Crafty.pixelart`, do we have that would be good candidates for this?

**Comment by starwed** [comment]
> I don't understand difference between top-level or nested options. Is there an implementation caveat when trying to merge nested options?

If we only allow options like this: 

```
options = {
    scaleX: 5,
    scaleY: 7,
}
```

then each option can use a getter/setter to add logic for triggering events or returning global default values. If instead we use

```
options = {
    scale: { x:5, y:7}
}
```

it gets more complicated.  (It also requires cloning objects when merging default and specific objects.)

> On the other hand, what options, beside Crafty.pixelart, do we have that would be good candidates for this?

We'd likely want a similar pattern for components, too -- for instance, it should be possible to globally set the g-factor of the gravity component, but still override it for specific entities.  A general pattern where referencing `entity.options.someOption` calls a getter that returns `this.options._someOption || global.options._someOption` would be pretty useful.

_e:_ Mostly, allowing only top-level properties on an options object would be simpler and give us greater flexibility for how we extend the notion.

**Comment by mucaho** [comment]
Makes sense (+1 for top-level options), thanks for explaining. This would be quite useful.
The _cascading_ options could be extended to the respective system / layer.
`this.options._someOption || this.layer.options_someOption || global.options._someOption`
